File: src/main/java/com/graqr/threshr/model/redsky/products/ProductClassification.java
package com.graqr.threshr.model.redsky.products;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record ProductClassification(
        @Nullable @JsonProperty("product_type") Long productType,
        @Nullable @JsonProperty("product_type_name") String productTypeName,
        @Nullable @JsonProperty("purchase_behavior") String purchaseBehavior,
        @Nullable @JsonProperty("item_type") ItemType itemType) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/ProductClassification.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Nullable
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class ProductClassification(
    @param:Nullable @param:JsonProperty("product_type") val productType: Long,
    @param:Nullable @param:JsonProperty("product_type_name") val productTypeName: String,
    @param:Nullable @param:JsonProperty("purchase_behavior") val purchaseBehavior: String,
    @param:Nullable @param:JsonProperty("item_type") val itemType: ItemType
) {}
